### PR TITLE
Restored compatibility with Orocos KDL after Joint.None was removed

### PR DESCRIPTION
**Pull request by fjulian**
https://github.com/orocos/orocos_kinematics_dynamics/commit/d518c8b86e2be19fc1389fda91c9f566c21c0b8c removed attribute `PyKDL.Joint.None`. With this PR, compatibility is ensured, preferring `PyKDL.Joint.Fixed` and only falling back to `PyKDL.Joint.None` if the former is no attribute of `PyKDL.Joint`. See discussion in https://github.com/ros/kdl_parser/issues/44.

### CODE REVIEW

**Review by clalancette — APPROVED** [review]
Looks good to me, thanks for picking this up.  I'm going to approve, but since I had a hand in developing it, I'd rather get a second review before merging.

@sloretz Could you take a look when you get a chance?

**Review by sloretz — APPROVED** [review]

**Comment by sloretz** [comment]
Thanks for the PR! LGTM and the PR job passed, so I'll merge this.

@fjulian since you're building from source, does that mean it's ok if this isn't released right away?

**Comment by clalancette** [comment]
> Thanks for the PR! LGTM and the PR job passed, so I'll merge this.

Thanks, appreciated.  I'd say that we also need to port this to ROS 2, but it turns out we haven't ported `kdl_parser_py` to ROS 2 at all.  It actually looks like an easy port, except for the tests, but that's a task for another day.  I'll at least open an issue about that.

**Comment by fjulian** [comment]
Thanks for merging!

> does that mean it's ok if this isn't released right away?

Yes of course, it can wait from my side.

**Pull request merged**
